apps/bluetooth-music/app.js

```diff
@@ -0,0 +1,98 @@
+'use strict'
+
+var logger = require('logger')('bluetooth-music')
+var Application = require('@yodaos/application').Application
+var bluetooth = require('@yoda/bluetooth')
+var protocol = bluetooth.protocol
+var AudioFocus = require('@yodaos/application').AudioFocus
+var audioFocus = null
+var a2dp = null
+
+function onAudioFocusGained () {
+  logger.debug('onAudioFocusGained')
+  a2dp.unmute()
+}
+
+function onAudioFocusLost () {
+  logger.debug('onAudioFocusLost')
+  a2dp.mute()
+  app.finalize()
+}
+
+function onAudioStateChangedListener (mode, state, extra) {
+  logger.debug(`${mode} onAudioStateChanged(${state})`)
+  switch (state) {
+    case protocol.AUDIO_STATE.PLAYING:
+      break
+    case protocol.AUDIO_STATE.PAUSED:
+    case protocol.AUDIO_STATE.STOPPED:
+      audioFocus.abandon()
+      break
+    case protocol.AUDIO_STATE.VOLUMN_CHANGED:
+      break
+  }
+}
+
+var app = Application({
+  created: () => {
+    logger.debug('created')
```

Comment:
Could this logs be removed?

Reply:
As discuss, the log is for trace, and will be auto removed in release version.